### PR TITLE
update user

### PR DESCRIPTION
**Pull request by puteriffah**

### CODE REVIEW

**Comment by puteriffah** [comment]
ready

File: app/helpers/sessions_helper.rb [hunk 1]
```diff
@@ -14,7 +14,12 @@ def current_user
       @current_user ||= User.find_by id: session[:user_id]
     end
   end
-  
+
+  def current_user?(user)
+    user && user == current_user
+  end
+    
+
```
Comment (ngocvt-0484) [inline app/helpers/sessions_helper.rb:25]: remove this line

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -20,6 +27,30 @@ def create
 
   def show; end
 
+  def edit
+    @user = User.find params[:id]
```
Comment (ngocvt-0484) [inline app/controllers/users_controller.rb:31]: define function load user and use before_action to call it
```
before_action :load_user, only: [:show, :edit, :update]
def load_user
    @user = User.find_by id: params[:id]
end
```

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -20,6 +27,30 @@ def create
...
+  end
+
+  def update
+    @user = User.find_by id: params[:id]
```
Comment (ngocvt-0484) [inline app/controllers/users_controller.rb]: remove

File: app/controllers/users_controller.rb (hunk 2 above)
Comment (ngocvt-0484) [inline app/controllers/users_controller.rb]: remove it because you loaded user by before_action :load_user

File: app/views/users/new.html.erb [hunk 4]
```diff
@@ -1,19 +1,9 @@
...
-      <%= f.text_field :name, class: "form-control" %> 
-      <%= f.label :email, "Email" %>
-      <%= f.email_field :email, class: "form-control" %>
-      <%= f.label :password, "Password" %>
-      <%= f.password_field :password, class: "form-control" %>
-      <%= f.label :password_confirmation, t(".confirmation") %>
-      <%= f.password_field :password_confirmation, class: "form-control" %>
-      <%= f.submit t(".signup_create"), class: "btn btn-primary" %>
-    <% end %>
+    <%= render "form"%>
```
Comment (ngocvt-0484) [inline app/views/users/new.html.erb]:
```suggestion
    <%= render "form" %>
```

**Comment by puteriffah** [comment]
ready

File: app/helpers/sessions_helper.rb (hunk 1 above)
Comment (puteriffah) [reply app/helpers/sessions_helper.rb:25]: done

**Pull request merged**